app/main.py

```diff
@@ -1,8 +1,41 @@
 class Car:
-    # write your code here
-    pass
+    def __init__(self, comfort_class: int,
+                 clean_mark: int, brand: str) -> None:
+        self.comfort_class = comfort_class
+        self.clean_mark = clean_mark
+        self.brand = brand
 
 
 class CarWashStation:
-    # write your code here
-    pass
+    def __init__(self, distance_from_city_center: float, clean_power: int,
+                 average_rating: float, count_of_ratings: int) -> None:
+        self.distance_from_city_center = distance_from_city_center
+        self.clean_power = clean_power
+        self.average_rating = average_rating
+        self.count_of_ratings = count_of_ratings
+
+    def serve_cars(self, cars: list[Car]) -> float:
+        total_income = 0
+
+        for car in cars:
+            if car.clean_mark < self.clean_power:
+                total_income += self.calculate_washing_price(car)
+                self.wash_single_car(car)
+
+        return total_income
+
+    def calculate_washing_price(self, car: Car) -> float:
+        price = ((car.comfort_class * (self.clean_power - car.clean_mark)
+                  * self.average_rating) / self.distance_from_city_center)
+        price = round(price, 1)
+
+        return price
+
+    def wash_single_car(self, car: Car) -> None:
+        car.clean_mark = self.clean_power
+
+    def rate_service(self, new_rating: int) -> None:
+        self.count_of_ratings += 1
+        self.average_rating = round((
+            (self.average_rating * (self.count_of_ratings - 1)
+             + new_rating) / self.count_of_ratings), 1)
```

Comment:
The calculation of `average_rating` should ensure that the division is performed correctly. Since `self.count_of_ratings` is incremented before this calculation, the formula seems correct, but ensure that `new_rating` is always a valid integer.